rnd/autogpt_server/autogpt_server/schema.prisma

```diff
@@ -0,0 +1,93 @@
+datasource db {
+  provider = "sqlite"
+  url      = "file:./database.db"
+}
+
+generator client {
+  provider             = "prisma-client-py"
+  recursive_type_depth = 5
+  interface            = "asyncio"
+}
+
+// This model describe a single node in the Agent Graph/Flow (Multi Agent System).
+model AgentNode {
+  id String @id
+
+  agentComponentId String
+  AgentComponent   AgentComponent @relation(fields: [agentComponentId], references: [id])
+
+  // On which produced output, that the child node component should be executed.
+  // The length has to be equal to ChildNodes(ChildNodes[i]) will be triggered by trigger[i]).
+  // Or optionally equal to zero, which any types output will trigger the children.
+  childNodeOutputTrigger AgentComponentOutput[]
+
+  // These fields define the node relations as a graph.
+  parentNodeId String?
+  ParentNode   AgentNode?  @relation("AgentGraph", fields: [parentNodeId], references: [id])
+  ChildNodes   AgentNode[] @relation("AgentGraph")
+
+  ExecutionHistory AgentNodeExecution[]
+}
+
+// This model describe a component that will be executed by the AgentNode.
+model AgentComponent {
+  id   String @id
+  name String
+  code String
+
+  // We allow a component to have multiple types of output.
+  Output      AgentComponentOutput[]
```

Comment:
I think by default the agent (AutoGPT) will only have one output, so maybe we can by default set one output entry with outputName "output", and outputSchema "str" ?

Reply:
IMO we need to support control flow outputs that have unique schemas between control flows

Reply:
AutoGPT won't be directly plugged into the graph but will need a wrapper, so the AutoGPT output doesn't really matter here.

Reply:
Each AgentComponentOutput will have different schema